lib/kadira.js

```diff
@@ -212,23 +212,49 @@ Kadira._send = function (payload, callback) {
 
   function callHTTP() {
     Kadira._postData(endpoint, httpOptions, function(err, response){
-      if(response && response.statusCode === 401) {
-        // do not retry if authentication fails
-        throw new Error('Kadira: AppId, AppSecret combination is invalid');
+      if(!(response && response.statusCode)) {
+        err = err || new Error("Kadira: Unknown Error");
+        return tryAgain(err);
       }
 
-      if(response && response.statusCode == 200) {
+      var status = response.statusCode;
+
+      if(status === 200) {
         if(payloadRetries > 0) {
           logger('connected again and payload sent.')
         }
+
         cleaPayloadRetry();
-        callback && callback();
-      } else {
+        return callback && callback();
+      }
+
+      // do not retry if authentication fails
+      if(status === 401) {
+        throw new Error('Kadira: AppId, AppSecret combination is invalid');
+      }
+
+      if(status < 400 || status > 600) {
+        err = err || new Error("Unknown error: " + status);
+        return tryAgain(err);
+      }
+
+      if(status >= 500 && status < 600) {
         if(!err) {
-          err = new Error("Status code: " + response.statusCode);
+          err = new Error("Kadira: Unknown server error: " + status);
         }
-        tryAgain(err);
+
+        return tryAgain(err);
+      }
+
+      // do not retry for kadira agent errors
+      if(status === 429 || status === 430) {
```

Comment:
We don't need to tell which kind of error here. Just printing the statusCode is enough. 

The we can have another page with known errors. 401 is a special case.